### DISCUSSION TITLE
Question about loading from checkpoint

### DISCUSSION THREAD

**Discussion by wangq326** [SalesforceAIResearch/uni2ts#64] [General]
Hello，

I am trying to run your [example/moirai_forecast_pandas.ipynb](https://github.com/SalesforceAIResearch/uni2ts/tree/main/example/moirai_forecast_pandas.ipynb) script, but when I try to load the checkpoint with the following statement
```
model = MoiraiForecast(
    module=MoiraiModule.from_pretrained(f"loacalfoler/moirai-1.0-R-small"),
    prediction_length=48,
    context_length=168,
    patch_size=32,
    num_samples=100,
    target_dim=1,
    feat_dynamic_real_dim=0,
    past_feat_dynamic_real_dim=0,
)
```
I encountered an error `AttributeError: type object 'MoiraiModule' has no attribute 'from_pretrained'`. I checked the current code, in the `module.py` script,
```
class MoiraiModule(
    nn.Module,
    PyTorchModelHubMixin,
    coders={DistributionOutput: (encode_distr_output, decode_distr_output)},
):
    def __init__(...):
        ...
    def forward(...):
        ...
```
and in the `forecast.py` script:
```
class MoiraiForecast(L.LightningModule):
    ...
```
Since I am not familiar with the `L.LightningModule` class, I am unsure whether it is appropriate to directly add `L.LightningModule` to the `MoiraiModule` inheritance, and whether it might cause other interference?

In addition, if I mimic the previous example script and execute
```
model = MoiraiForecast.load_from_checkpoint(
    checkpoint_path=f"localfolder/Moirai-1.0-R-small/model.ckpt",
    prediction_length=48,
    context_length=168,
    patch_size=32,
    num_samples=100,
    target_dim=1,
    feat_dynamic_real_dim=0,
    past_feat_dynamic_real_dim=0,
    map_location="cuda:0" if torch.cuda.is_available() else "cpu",
)
```
then the checkpoint can be loaded successfully.

[This issue](https://github.com/SalesforceAIResearch/uni2ts/issues/40) discussed the relevant content, but the error messages are not consistent.

**Comment by gorold** [comment]
2 things to do:
1. pull the most recent commit
2. re-install uni2ts module / update the huggingface_hub and safetensors packages
